kafka/consumer.go

```diff
@@ -150,6 +150,18 @@ func (cfg *ConsumerConfig) finalize() error {
 	if cfg.FetchMinBytes < 0 {
 		errs = append(errs, errors.New("kafka: fetch min bytes cannot be negative"))
 	}
+	if cfg.MaxPollBytes > 0 {
+		// math.MaxInt32 is 1<<31-1.
+		if cfg.MaxPollBytes > 1<<30 {
+			cfg.MaxPollBytes = 1 << 30
+		}
+		if cfg.BrokerMaxReadBytes == 0 {
+			cfg.BrokerMaxReadBytes = cfg.MaxPollBytes * 2
+		}
+	}
+	if cfg.BrokerMaxReadBytes < 0 || cfg.BrokerMaxReadBytes > 1<<30 {
+		cfg.BrokerMaxReadBytes = 1 << 30
```

Comment:
My rationale was that it's useful to have this limited to the max amount, since it's quite large. But perhaps we should log an info message that this is being done.

Reply:
+1 on setting to the max and logging a warning. But in alignment with @kruskall 's concerns, I don't see why this should be treated different than `FetchMinBytes`, `MaxPollBytes` and `MaxPollPartitionBytes` (https://github.com/elastic/apm-queue/pull/606/files#diff-bd03622ad29aac2f6cffdf1c2755ccffd0981f3e4ffef219508a265509f4fe8eR144-R152) - should we also set these to `0` if negative, and log warnings?

Reply:
@marclop I'm not sure I follow your approach to the latest change. What I tried to suggest in the above message is to go further with your approach and also align the existing code to issue log messages and set config values to the franz-kafka minimum or maximum for existing config options such as `FetchMinBytes`, `MaxPollBytes` and `MaxPollPartitionBytes`. I did not mean to suggest to issue an error for negative settings.

Reply:
I can do `MaxPollBytes` and `MaxPollPartitionBytes`. But `FetchMinBytes` doesn't really have an upper limit. We can set one arbitrarily if we wish, but I'd rather not.

Reply:
Added in [0544375](https://github.com/elastic/apm-queue/pull/606/commits/054437553ec91e5d976c308f1a2382b465a4978d)